### PR TITLE
SDK Changes Refactor

### PR DESCRIPTION
**Pull request by Velin92**
This PR includes the two following refactors coming from SDK changes:

 - App Service to control at the same time the encryption sync AND the room list API
 - Internal identifiers to keep a stable identifiction system for each timeline item, this is independent from the event id which can be not available in some cases (for example between the transition from Local echo to Remote Message)
 - Refactor to properly use the right identifier for the right call, and joined the identifiers into a single  Hashable struct
 - SDK Bump to 1.0.98

### CODE REVIEW

File: ElementX/Sources/Services/Client/ClientProxy.swift [hunk 1]
```diff
@@ -385,85 +372,73 @@ class ClientProxy: ClientProxyProtocol {
...
+            let roomListService = appService.roomListService()
+
+            roomSummaryProvider = RoomSummaryProvider(roomListService: roomListService,
+                                                      eventStringBuilder: RoomEventStringBuilder(stateEventStringBuilder: RoomStateEventStringBuilder(userID: userID)),
+                                                      name: "AllRooms")
+
+            try await roomSummaryProvider?.setRoomList(roomListService.allRooms())
+
+            inviteSummaryProvider = RoomSummaryProvider(roomListService: roomListService,
+                                                        eventStringBuilder: RoomEventStringBuilder(stateEventStringBuilder: RoomStateEventStringBuilder(userID: userID)),
```
Comment (pixlwave) [inline ElementX/Sources/Services/Client/ClientProxy.swift:396]: I realise this is how it was, but should we make this string builder once and pass it to both providers?

File: ElementX/Sources/Services/Room/RoomSummary/RoomSummaryProvider.swift [hunk 2]
```diff
@@ -140,7 +140,23 @@ class RoomSummaryProvider: RoomSummaryProviderProtocol {
         
         MXLog.verbose("\(name): Finished applying \(diffs.count) diffs, new room list \(rooms.compactMap { $0.id ?? "Empty" })")
     }
-        
+
+    private func fetchLastMessage(roomListItem: RoomListItemProtocol) -> EventTimelineItem? {
+        class RoomMessageBox {
```
Comment (pixlwave) [inline ElementX/Sources/Services/Room/RoomSummary/RoomSummaryProvider.swift:145]: Box feels very Rust and a bit alien to Swift imo…

File: ElementX/Sources/Services/Client/ClientProxy.swift [hunk 3]
```diff
@@ -385,85 +372,73 @@ class ClientProxy: ClientProxyProtocol {
...
+
+            self.appService = appService
+            self.roomListService = roomListService
+
+            appServiceUpdateTaskHandle = appService.state(listener: AppStateObserverProxy { [weak self] state in
                 guard let self else { return }
+                MXLog.info("Received app service update: \(state)")
+                switch state {
+                case .error:
+                    restartSync()
```
Comment (pixlwave) [inline ElementX/Sources/Services/Client/ClientProxy.swift:407]: Ohhhh interesting, I'd wondered how the app was retrying requests in a quick loop. In the future we should be adding some kind of back-off on this.

**Review by pixlwave — APPROVED** [review]
Very nice, this is a really positive change to the code :)

File: ElementX.xcodeproj/xcshareddata/xcschemes/ElementX.xcscheme
[the remark this answers is not in the document]
Comment (Velin92) [reply ElementX.xcodeproj/xcshareddata/xcschemes/ElementX.xcscheme:1]: actually this was in @aringenbach 's PR and xcodegen does not revert it back, is it actually needed?
(I actually realised that when I compiled the code the proj was automatically doing this update too to the scheme, so I guess is probably related to Xcode updtes?)

File: ElementX/Sources/Services/Room/RoomSummary/RoomSummaryProvider.swift (hunk 2 above)
Comment (Velin92) [reply ElementX/Sources/Services/Room/RoomSummary/RoomSummaryProvider.swift:145]: this was originally in @aringenbach 's PR and I think is related to the async method solution.
Maybe we want to call it `EventTimelineItemReference` ?

File: ElementX.xcodeproj/xcshareddata/xcschemes/ElementX.xcscheme
[the remark this answers is not in the document]
Comment (pixlwave) [reply ElementX.xcodeproj/xcshareddata/xcschemes/ElementX.xcscheme:1]: Oh ok well if xcodegen doesn't revert it, then its fine.

File: ElementX/Sources/Services/Room/RoomSummary/RoomSummaryProvider.swift (hunk 2 above)
Comment (pixlwave) [reply ElementX/Sources/Services/Room/RoomSummary/RoomSummaryProvider.swift:145]: Given its nested could simply be `FetchResult` or similar.

File: ElementX/Sources/Screens/RoomScreen/RoomScreenViewModel.swift [hunk 4]
```diff
@@ -449,17 +449,20 @@ class RoomScreenViewModel: RoomScreenViewModelType, RoomScreenViewModelProtocol
             setComposerMode(.edit(originalItemId: messageTimelineItem.id))
         case .copyPermalink:
             do {
-                let permalink = try PermalinkBuilder.permalinkTo(eventIdentifier: eventTimelineItem.id, roomIdentifier: timelineController.roomID,
+                guard let eventID = eventTimelineItem.id.eventID else {
+                    break
```
[the remark this answers is not in the document]
Comment (Velin92) [reply ElementX/Sources/Screens/RoomScreen/RoomScreenViewModel.swift:453]: or even better I can just do:
```
 displayError(.alert(L10n.errorFailedCreatingThePermalink))
 break
 ```

File: ElementX/Sources/Services/Timeline/TimelineController/RoomTimelineController.swift [hunk 5]
```diff
@@ -121,68 +121,81 @@ class RoomTimelineController: RoomTimelineControllerProtocol {
...
+    func sendMessage(_ message: String, inReplyTo itemID: TimelineItemIdentifier?) async {
         if itemID == nil {
             MXLog.info("Send message in \(roomID)")
         } else {
             MXLog.info("Send reply in \(roomID)")
         }
-        
-        switch await roomProxy.sendMessage(message, inReplyTo: itemID) {
+
+        switch await roomProxy.sendMessage(message, inReplyTo: itemID?.eventID) {
```
[the remark this answers is not in the document]
Comment (Velin92) [reply ElementX/Sources/Services/Timeline/TimelineController/RoomTimelineController.swift:131]: done

**Pull request merged**